src/modules/avformat/producer_avformat.c

```diff
@@ -1833,7 +1857,35 @@ static int producer_get_image( mlt_frame frame, uint8_t **buffer, mlt_image_form
 					codec_context->reordered_opaque = int_position;
 					if ( int_position >= req_position )
 						codec_context->skip_loop_filter = AVDISCARD_NONE;
-					ret = avcodec_decode_video2( codec_context, self->video_frame, &got_picture, &self->pkt );
+					
+					char errstr[1000];
+
+					if ( !self->video_frame )
+						self->video_frame = av_frame_alloc();
+					else
+						av_frame_unref( self->video_frame );
+
+#ifdef USE_VAAPI
+					if ( !self->sw_video_frame )
+						self->sw_video_frame = av_frame_alloc();
+					else
+						av_frame_unref( self->sw_video_frame );
+#endif
+
+					do 
+					{
+						if ( ( ret = avcodec_send_packet( codec_context, &self->pkt ) ) < 0 )
+						{
+							// av_strerror( ret, errstr, 1000 );
+							// mlt_log_error( NULL, "avcodec_send_packet() failed %s\n", errstr );
+						}
+						if ( ( ret = avcodec_receive_frame( codec_context, self->video_frame ) ) < 0 ) 
+						{
+							// av_strerror( ret, errstr, 1000 );
+							// mlt_log_error( NULL, "avcodec_receive_frame() failed %s\n", errstr );
+						}
+					} while ( ret == AVERROR( EAGAIN ) );
```

Comment:
Some decoders need multiple packets before they will return a frame. I will share my branch with you where I have adopted this API.